templates/new_structure/DeployStage.yml

```diff
@@ -0,0 +1,48 @@
+parameters:
+- name: environment
+  type: string
+- name: tfVersion
+  type: string
+- name: workingDirectory
+  type: string
+- name: serviceConnection
+  type: string
+- name: backendResourceGroupName
+  type: string
+- name: backendStorageAccountName
+  type: string
+- name: backendContainerName
+  type: string
+
+stages:
+- stage: apply
+  condition: |
+    and (
+      succeeded(),
+      in(dependencies.build.result, 'Succeeded')
+    )
+  dependsOn:
+    - "build"
+  displayName: "Apply"
```

Comment:
We might need support for private agents - lets default to using the azure provided or default, but the option to use private agents.

Reply:
Fixed in new commits.